hibernate-core/src/main/java/org/hibernate/id/enhanced/SequenceStyleGenerator.java

```diff
@@ -280,6 +282,21 @@ private int adjustIncrementSize(
 		return determineAdjustedIncrementSize( optimizationStrategy, incrementSize );
 	}
 
+	private boolean isSchemaToBeRecreated(ConfigurationService configurationService) {
+		Action action = configurationService.getSetting( AvailableSettings.JAKARTA_HBM2DDL_DATABASE_ACTION, Action.class, null );
+		if ( action == null ) {
+			String setting = configurationService.getSetting( AvailableSettings.JAKARTA_HBM2DDL_DATABASE_ACTION, StandardConverters::asString);
+			if ( setting != null ) {
+				action = Action.interpretJpaSetting( setting );
+			}
+			else {
+				setting  = configurationService.getSetting( AvailableSettings.HBM2DDL_AUTO, StandardConverters::asString );
+				action = Action.interpretHbm2ddlSetting( setting );
+			}
+		}
+		return action == Action.CREATE || action == Action.CREATE_DROP;
+	}
+
```

Comment:
This implementation looks fine to me. Only note, you're not considering contributor-specific settings (`Contributable#getContributor`): see how `SchemaManagementToolCoordinator.ActionGrouping#interpret` reads the settings based on `contributor`s, and falls back to the default values (the ones you're already reading here) - not as useful for our tests maybe but might as well included all cases.

Reply:
We'd always be looking first for a setting such as e.g. `jakarta.persistence.schema-generation.database.action.orm` before going after the setting that might actually have a value (i.e. without the '.orm' appendix). Not sure if that makes any sense tbh?

Reply:
> before going after the setting that might actually have a value (i.e. without the '.orm' appendix)

But what if the contributor-specific setting is set and the generic one is not? Contributor settings take precedence, and while this is not the case in our test-suite might as well handle it so this optimization works in all cases.

See how the method `org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.ActionGrouping#interpret(org.hibernate.boot.Metadata, java.util.Map<?,?>)` determines the action to use - we should first check the setting with the specific contributor for the generator's sequence (returned by `org.hibernate.id.enhanced.SequenceStyleGenerator#determineContributor`), and then fallback to the generic one.